### DISCUSSION TITLE
Output a sequence of messages from a single "user script"

### DISCUSSION THREAD

**Discussion by luisrayas3** [foxglove/community#883] [Q&A]
My motivating use-case is to take a trajectory message, which has an array of points, and expand it into a message per-point to be plotted in a plot. I can imagine similar use-cases where a sequence of values is encoded in a single message and wants to be expanded in a plot. This could also provide a work-around for the lack of a 2D-plot panel.

**Comment by defunctzombie** [comment]
Thanks for the feedback.

The plot panel does support plotting points from a single message using the [Index](https://docs.foxglove.dev/docs/visualization/panels/plot/#index) feature. Give that a try and it might work for your trajectory messages.

**Comment by luisrayas3** [comment]
Thanks for the help! This is a partial solution/workaround. It does help me plot multiple points from the same message (actually using "Path (current)" was closer to what I wanted) but there are still 2 limitations:

1. Cannot plot with other time-series in the same plot. In my case the x-axis is still time, even though it's one I compute. I would like to visualize these in the same plot as other time-series. With this solution I need to split them across different plots
2. I can't get lines connecting the points to show up. This seems to be a problem both with "Index" and "Path (...)" for the x-axis.
